test/EnvironmentManagerTests/Helper.h

```diff
@@ -0,0 +1,167 @@
+
+    // +1 for null character
+    const DWORD NAME_LENGTH = sizeOfLongestNameInCharacters + 1;
+    const DWORD VALUE_SIZE_IN_BYTES = sizeOfLongestValueInCharacters * sizeof(WCHAR);
+
+    std::unique_ptr<wchar_t[]> environmentVariableName(new wchar_t[NAME_LENGTH]);
+    std::unique_ptr<BYTE[]> environmentVariableValue(new BYTE[VALUE_SIZE_IN_BYTES]);
+
+    for (int valueIndex = 0; valueIndex < numberOfValues; valueIndex++)
+    {
+        DWORD nameLength = NAME_LENGTH;
+        DWORD valueSize = VALUE_SIZE_IN_BYTES;
+        LSTATUS enumerationStatus = RegEnumValueW(hKey
+            , valueIndex
+            , environmentVariableName.get()
+            , &nameLength
+            , nullptr
+            , nullptr
+            , environmentVariableValue.get()
+            , &valueSize);
+
+        // An empty name indicates the default value.
+        if (nameLength == 0)
+        {
+            continue;
+        }
+
+        // If there was an error getting the value
+        if (enumerationStatus != ERROR_SUCCESS && enumerationStatus != ERROR_NO_MORE_ITEMS)
+        {
+            THROW_HR(HRESULT_FROM_WIN32(enumerationStatus));
+        }
+
+        environmentVariables.Insert(environmentVariableName.get(), reinterpret_cast<LPWSTR>(environmentVariableValue.get()));
+
+        environmentVariableName.reset(new wchar_t[NAME_LENGTH]);
+        environmentVariableValue.reset(new BYTE[VALUE_SIZE_IN_BYTES]);
+    }
+
+    return environmentVariables.GetView();
+}
+
+inline EnvironmentVariables GetEnvironmentVariablesForUser()
+{
+    wil::unique_hkey environmentVariablesHKey;
+    THROW_IF_FAILED(HRESULT_FROM_WIN32(RegOpenKeyEx(HKEY_CURRENT_USER, USER_EV_REG_LOCATION, 0, KEY_READ, environmentVariablesHKey.addressof())));
+    return GetEnvironmentVariablesFromRegistry(environmentVariablesHKey.get());
+}
+
+inline EnvironmentVariables GetEnvironmentVariablesForMachine()
+{
+    wil::unique_hkey environmentVariablesHKey;
+    THROW_IF_FAILED(HRESULT_FROM_WIN32(RegOpenKeyEx(HKEY_LOCAL_MACHINE, MACHINE_EV_REG_LOCATION, 0, KEY_READ, environmentVariablesHKey.addressof())));
+    return GetEnvironmentVariablesFromRegistry(environmentVariablesHKey.get());
+}
+
+inline EnvironmentVariables GetEnvironmentVariablesForProcess()
+{
+    LPWSTR environmentVariablesBlock = GetEnvironmentStrings();
+    VERIFY_IS_NOT_NULL(environmentVariablesBlock);
+
+    LPTSTR variable;
+    variable = (LPTSTR)environmentVariablesBlock;
+
+    StringMap environmentVariables;
+    while (*variable)
+    {
+        std::wstring environmentVariable(variable);
+        size_t locationOfNull = environmentVariable.find_last_of(L'=');
+
+        environmentVariables.Insert(environmentVariable.substr(0, locationOfNull), environmentVariable.substr(locationOfNull + 1));
+
+        variable += lstrlen(variable) + 1;
+    }
+
+    VERIFY_WIN32_BOOL_SUCCEEDED(FreeEnvironmentStrings(environmentVariablesBlock));
+
+    return environmentVariables.GetView();
+}
+
+/// <summary>
```

Comment:
Blech. XML doccomments.

I'm partial to [Doxygen](https://www.doxygen.nl/index.html] with Javadoc-style notation. Cheatsheet for 99%+ of all your doccomment needs:

```
/// First sentence is your brief description just like Javadoc. You can
/// have more verbiage too for a full description if one sentence isn't
/// enough.
/// @param paramname description of parameter
/// @return description of return value
/// @exception MyExceptionObject description of the exception that could be thrown
/// @see xref to one or more class, function, method, variable, file or URL
/// @note message of interest
/// @warning more concerning message of interest
/// @todo description of something to do
and 3+ tilde (~) to start a code block (great for samples). Ends with another line with 3+ tilde (~)
```

Equivalent information but a lot less noise. Plus reading the comment in source becomes reasonably legible in its own right

Reply:
I see.  Yeah, the XML styled comments came from my C# days.